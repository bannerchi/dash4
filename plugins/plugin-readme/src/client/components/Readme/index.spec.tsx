/// <reference types="@types/jest" />
const useFullScreenMock = jest.fn().mockImplementation(() => ({
	fullscreen: false,
	enableFullscreen: jest.fn(),
	disableFullscreen: jest.fn(),
}));
jest.mock('../../hooks', () => ({
	__esModule: true, // this property makes it work
	// default: 'mockedDefaultExport',
	useMarkdownData: jest.fn().mockImplementation(() => '<div>markdown</div>'),
	useFullscreen: useFullScreenMock,
}));

import React from 'react';
import { render } from 'react-testing-library';
import { Readme } from '.';

describe('PluginReadme', () => {
	describe('Readme', () => {
		test('exist', () => {
			expect(typeof Readme).toBe('object');
		});
		test('render default', () => {
			const { container } = render(
				<Readme
					id={'i1-root'}
					name={'PluginReadme'}
					lowerCaseName={'plugin-readme'}
					clientConfig={{
						file: '/path-to-file',
					}}
				/>
			);
			expect(container.firstChild).toMatchSnapshot();
		});
		test('render fullscreen', () => {
			useFullScreenMock.mockReturnValue({
				fullscreen: true,
				enableFullscreen: jest.fn(),
				disableFullscreen: jest.fn(),
			});
			const { container } = render(
				<Readme
					id={'i1-root'}
					name={'PluginReadme'}
					lowerCaseName={'plugin-readme'}
					clientConfig={{
						file: '/path-to-file',
					}}
				/>
			);
			expect(container.firstChild).toMatchSnapshot();
			jest.restoreAllMocks();
		});
	});
});
