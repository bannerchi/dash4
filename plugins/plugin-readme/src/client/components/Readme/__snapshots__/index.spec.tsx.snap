// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginReadme Readme render default 1`] = `
<div
  class="Component-window-0-1-9 Component-windowBright-0-1-11 Component-windowAutoStretch-0-1-10"
  tabindex="1"
>
  <div
    class="ReadmeRaw-windowBody-0-1-1 Component-body-0-1-14"
  >
    <div
      class="ReadmeRaw-markdownWrapper-0-1-3 ReadmeRaw-markdownWrapper-0-1-8"
    >
      <div
        class="markdown-body"
      >
        <div>
          markdown
        </div>
      </div>
    </div>
    <div
      class="ReadmeRaw-buttonWrapper-0-1-5"
    >
      <button
        class="ReadmeRaw-showMoreButton-0-1-6 btn btn-outline-primary btn-sm"
        type="button"
      >
        show more
      </button>
    </div>
  </div>
</div>
`;

exports[`PluginReadme Readme render fullscreen 1`] = `
<div
  class="Component-window-0-1-9 Component-windowBright-0-1-11 Component-windowAutoStretch-0-1-10"
  tabindex="1"
>
  <div
    class="ReadmeRaw-windowBody-0-1-1 Component-body-0-1-14"
  >
    <div
      class="ReadmeRaw-markdownWrapper-0-1-3 ReadmeRaw-markdownWrapper-0-1-15"
    >
      <div
        class="markdown-body"
      >
        <div>
          markdown
        </div>
      </div>
    </div>
    <div
      class="ReadmeRaw-buttonWrapper-0-1-5"
    >
      <button
        class="ReadmeRaw-showMoreButton-0-1-6 btn btn-outline-primary btn-sm"
        type="button"
      >
        show more
      </button>
    </div>
  </div>
</div>
`;
